FindOpenStudioSDK.cmake

```diff
@@ -1,78 +1,148 @@
 set(OPENSTUDIO_VERSION_MAJOR 3)
-set(OPENSTUDIO_VERSION_MINOR 0)
-set(OPENSTUDIO_VERSION_PATCH 1)
+set(OPENSTUDIO_VERSION_MINOR 1)
+set(OPENSTUDIO_VERSION_PATCH 0)
 set(OPENSTUDIO_VERSION "${OPENSTUDIO_VERSION_MAJOR}.${OPENSTUDIO_VERSION_MINOR}.${OPENSTUDIO_VERSION_PATCH}")
 
-find_package(openstudio "${OPENSTUDIO_VERSION}" CONFIG)
+#If this is a release enter the SHA as "+79857912c4"
+#set(OPENSTUDIO_VERSION_SHA "+09b7c8a554")
+#If this is a pre-release enter the pre-release and SHA as "-rc1+79857912c4"
+set(OPENSTUDIO_VERSION_SHA "-alpha+fa9a77bf17")
+
+# Paths where the cmake-downloaded archives will be put
+set(OPENSTUDIO_ARCHIVE_DIR "${PROJECT_BINARY_DIR}/OpenStudio-${OPENSTUDIO_VERSION}")
+
+# If downloaded, we need the SHA to match. This block is here since we need "OPENSTUDIO_PLATFORM" anyways
+if(APPLE)
+  set(OPENSTUDIO_EXPECTED_HASH 4f9d9fcf1f384bfc7e32557fe0c9913f)
+  set(OPENSTUDIO_PLATFORM "Darwin")
+  set(OPENSTUDIO_EXT "tar.gz")
+elseif(UNIX)
+  set(OPENSTUDIO_EXPECTED_HASH a978690432bca4c4eac5b780c89ca461)
+  set(OPENSTUDIO_PLATFORM "Linux")
+  set(OPENSTUDIO_EXT "tar.gz")
+elseif(WIN32)
+  set(OPENSTUDIO_EXPECTED_HASH c5c3bf8922ac695d22f0c94f957f535c)
+  set(OPENSTUDIO_PLATFORM "Windows")
+  set(OPENSTUDIO_EXT "tar.gz")
+endif()
+
+
+set(OPENSTUDIO_ARCHIVE_BASENAME "OpenStudio-${OPENSTUDIO_VERSION}${OPENSTUDIO_VERSION_SHA}-${OPENSTUDIO_PLATFORM}")
+set(OPENSTUDIO_ARCHIVE_NAME "${OPENSTUDIO_ARCHIVE_BASENAME}.${OPENSTUDIO_EXT}"
+  CACHE STRING "Archive Name, with extension" FORCE)
+
+# See if we can find the openstudio package with the right short version (eg 3.1.0) directly
+# (either installed and in PATH, or if user supplied openstudio_DIR in the cmake command)
+find_package(openstudio "${OPENSTUDIO_VERSION}" CONFIG
+  PATHS
+    "${OPENSTUDIO_ARCHIVE_DIR}/${OPENSTUDIO_ARCHIVE_BASENAME}"
+  HINTS
+    "${OPENSTUDIO_ARCHIVE_DIR}"
+)
```

Comment:
Try to locate it directly. 

If not, proceed to download.

You'll get something like this:

```
CMake Warning at FindOpenStudioSDK.cmake:36 (find_package):
  Could not find a configuration file for package "openstudio" that is
  compatible with requested version "3.1.0".
  The following configuration files were considered but not accepted:
    /usr/local/openstudio-3.0.1/lib/cmake/openstudio/openstudioConfig.cmake, version: 3.0.1
    /usr/local/openstudioapplication-1.0.1/lib/cmake/openstudio/openstudioConfig.cmake, version: 3.0.1
Call Stack (most recent call first):
  CMakeLists.txt:18 (include)
-- Try Downloading OpenStudio SDK: https://openstudio-builds.s3.amazonaws.com/3.1.0/OpenStudio-3.1.0-alpha%2Bfa9a77bf17-Linux.tar.gz
-- Download failed status code = 22, message = "HTTP response code said error"
-- Try Downloading OpenStudio SDK: http://openstudio-ci-builds.s3-website-us-west-2.amazonaws.com/develop/OpenStudio-3.1.0-alpha%2Bfa9a77bf17-Linux.tar.gz
-- [download 0% complete]
-- [download 1% complete]
```